### PR TITLE
Fixed incorrect date format conversion on input (DateEditor)

### PR DESCRIPTION
**Pull request by wszymanski**
### Context

Issue #9681 describes use case which cause conversion from `7/1/22` to `01/07/2022` while the result should be equal to `07/01/2022`. The investigation has shown that problem was related to introduction of `ShortcutManager`, not as it was supposed at the beginning - related to update of `moment.js` (#9381, change introduced within the same release).

The problem has existed, because introduced `ShortcutManager` captured events on `window` object instead of `document` element. Events captured on `window` are triggered later. Thus, `change` event captured by `Pikaday` https://github.com/Pikaday/Pikaday/blob/21f676e70d688d18b265f2c12fc38e8457c20645/pikaday.js#L586-L600 has been overwriting cell's value at the wrong time. What's more, until `12.0.0` release Handsontable has been stopping propagation of event. Without this behaviour, the current code seems effect in some flickering. 

https://user-images.githubusercontent.com/141330/189110464-7bc037ae-7f22-460e-bda0-9ad14ad4d5e6.mp4

This PR change changes this behavior.

I've tried to add [an extra test](https://gist.github.com/wszymanski/0b4998f3aa54d8f9bd9b28d268eb4186), but it seems that simulation done by Jasmine isn't perfect. The `change` event isn't simulated properly.

### How has this been tested?

1. I tested case shown in the issue.
2. I tested other ways of closing a cell.
3. I tested whether opened `Pikaday` widget looks properly for chosen `dateFormat`

Screenshot 2022-09-08 at 13 20 22

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #9681 

### Affected project(s):
- [x] `handsontable`
- [ ] `@handsontable/angular`
- [ ] `@handsontable/react`
- [ ] `@handsontable/vue`
- [ ] `@handsontable/vue3`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: handsontable/src/editors/dateEditor/dateEditor.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ import { isFunctionKey } from '../../helpers/unicode';
 import 'pikaday/css/pikaday.css';
 
 export const EDITOR_TYPE = 'date';
+const group = 'closingDateEditor';
```
Comment (jansiegel) [inline handsontable/src/editors/dateEditor/dateEditor.js:12]: The `group` variable name is a little generic; could you change it to be more descriptive?

**Review by jansiegel — CHANGES_REQUESTED** [review]

File: handsontable/src/editors/dateEditor/dateEditor.js (hunk 1 above)
Comment (wszymanski) [reply handsontable/src/editors/dateEditor/dateEditor.js]: Changed within: https://github.com/handsontable/handsontable/pull/9878/commits/45ec21784bddcf2b3491b106e7e882faf539f2ca

**Review by jansiegel — APPROVED** [review]
LGTM 👍

**Review by jansiegel — CHANGES_REQUESTED** [review]
Missed this one - the linter fails here; you forgot to change the variable name later in the file.

https://github.com/handsontable/handsontable/pull/9878/files#diff-e0a0d81eeeedd01921d00d6732206f8545cf1d4208438b6fcb3f30f87fd071e8R148

**Comment by wszymanski** [comment]
> Missed this one - the linter fails here; you forgot to change the variable name later in the file.
> 
> [#9878 (files)](https://github.com/handsontable/handsontable/pull/9878/files#diff-e0a0d81eeeedd01921d00d6732206f8545cf1d4208438b6fcb3f30f87fd071e8R148)

Thank you. Changed.

**Review by jansiegel — APPROVED** [review]

**Pull request merged**